alea/parameters.py

```diff
@@ -291,9 +292,9 @@ def __getattr__(self, name: str) -> Parameter:
         Raises:
             AttributeError: If the attribute is not found.
         """
-        if name in self.parameters:
-            return self.parameters[name]
-        else:
+        try:
+            return super().__getattribute__('parameters')[name]
+        except KeyError:
```

Comment:
In previous code, we can not do `deepcopy` of `Parameters`' instance. Because when `deepcopy` iteratively go through all attributes of the instance, it will call for `__getattr__` inside `__getattr__`, at `return self.parameters[name]`. Then this becomes an endless loop.

Reply:
I think the objects should be deep copyable especially when debugging.

Reply:
Makes a lot of sense, thanks!